src/batch/src/executor/generic_exchange.rs

```diff
@@ -160,12 +168,43 @@ impl<C: BatchTaskContext> GenericExchangeExecutor<C> {
 }
 
 #[try_stream(boxed, ok = DataChunk, error = RwError)]
-async fn data_chunk_stream(mut source: ExchangeSourceImpl) {
+async fn data_chunk_stream(
+    mut source: ExchangeSourceImpl,
+    metrics: Option<Arc<BatchMetrics>>,
+    downstream_id: TaskId,
+) {
     loop {
         if let Some(res) = source.take_data().await? {
             if res.cardinality() == 0 {
                 debug!("Exchange source {:?} output empty chunk.", source);
             }
+            if metrics.is_some() {
```

Comment:
When it will be None?

Reply:
None indicates not collect metrics. Such as local mode will not collect metrics. I will add comment later.

Reply:
I was checking into your problem but not any meaningful findings yet.

BTW, actor level exchange should be more close to this PR: https://github.com/singularity-data/risingwave/pull/2906

Reply:
> Must use >1 compute node to trigger this and the duplicate 2->1 shown is different compute node.

Quote from #3696 but I'm not sure why...

Reply:
> > Must use >1 compute node to trigger this and the duplicate 2->1 shown is different compute node.
> 
> Quote from #3696 but I'm not sure why...

I guess because it implement metrics in remote_input and it only collect metrics cross compute node.
https://github.com/singularity-data/risingwave/blob/0f6a83c72c9df406a9e6669d4e90f80a0b4ddef1/src/stream/src/executor/exchange/input.rs#L109